OpenEdXMobile/src/main/java/org/edx/mobile/base/BaseFragment.java

```diff
@@ -1,31 +1,38 @@
 package org.edx.mobile.base;
 
-import android.Manifest;
 import android.app.Activity;
 import android.content.Context;
 import android.content.Intent;
 import android.content.res.Configuration;
-import android.os.Build;
 import android.os.Bundle;
 
-import androidx.annotation.NonNull;
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
 import androidx.fragment.app.Fragment;
 
 import com.google.android.material.snackbar.Snackbar;
 
 import org.edx.mobile.R;
 import org.edx.mobile.http.notifications.SnackbarErrorNotification;
-import org.edx.mobile.util.PermissionsUtil;
 
 public class BaseFragment extends Fragment {
-    public interface PermissionListener {
-        void onPermissionGranted(String[] permissions, int requestCode);
-
-        void onPermissionDenied(String[] permissions, int requestCode);
-    }
 
     private boolean isFirstVisit = true;
-    protected PermissionListener permissionListener;
+
+    protected ActivityResultLauncher<String> requestPermissionLauncher =
```

Comment:
instead of defining this method for every child class can we follow the previous approach and forward the data using callback?
right now this launcher is only used in MainTabsDashboardFragment.

Reply:
I have created this method for permissions that don't require a followup response. But if needed, I can move it to MainTabsDashboardFragment.